app/src/main/java/me/ccrama/redditslide/util/OnSingleClickListener.java

```diff
@@ -0,0 +1,55 @@
+package me.ccrama.redditslide.util;
+
+import android.util.Config;
+import android.util.Log;
+import android.view.View;
+
+/**
+ * Implementation of {@link View.OnClickListener} that ignores subsequent clicks that happen too quickly after the first one.<br/>
+ * To use this class, implement {@link #onSingleClick(View)} instead of {@link View.OnClickListener#onClick(View)}.
+ */
+public abstract class OnSingleClickListener implements View.OnClickListener {
+
+    private long mLastClickTime;
+    private static final long MIN_DELAY_MS = 300;
+    private static final String TAG = OnSingleClickListener.class.getSimpleName();
+
+    /**
+     * Called when a view has been clicked.
+     *
+     * @param v The view that was clicked.
+     */
+    public abstract void onSingleClick(View v);
+
+    @Override
+    public final void onClick(View v) {
+        final long lastClickTime = mLastClickTime;
+        final long now = System.currentTimeMillis();
```

Comment:
Avoid using System.currentTimeMillis() for this kinds of things, as it is based on the "wall clock" of the phone. Use a monotonic alternative such as SystemClock.uptimeMillis()
See the "Class Overview" in https://developer.android.com/reference/android/os/SystemClock.html for more details.